test/test_layout.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include <iostream>
+#include <numeric>
+
+#include "doctest/doctest.h"
+
+#include "sparrow/layout.hpp"
+
+namespace sparrow
+{
+    static_assert(std::ranges::range<primitive_layout<uint8_t>>);
```

Comment:
Consider adding a `TODO` to test all the other base types (I'm working on maknig this easy, so that would have to wait for #15)